### PR TITLE
kube: `configmap` volume should be reused if already exists

### PR DESCRIPTION
**Pull request by flouthoc**
`podman play kube` creates a new volume for configmap, if same configmap
is applied again volume can be re-used, there is no need to remove and
re-create the volume again

Closes: https://github.com/containers/podman/issues/13715

### CODE REVIEW

File: pkg/domain/infra/abi/play.go [hunk 1]
```diff
@@ -290,7 +290,16 @@ func (ic *ContainerEngine) playKubePod(ctx context.Context, podName string, podY
 		if v.Type == kube.KubeVolumeTypeConfigMap && !v.Optional {
 			vol, err := ic.Libpod.NewVolume(ctx, libpod.WithVolumeName(v.Source))
 			if err != nil {
-				return nil, errors.Wrapf(err, "cannot create a local volume for volume from configmap %q", v.Source)
+				if strings.Contains(err.Error(), "already exists") {
```
Comment (Luap99) [inline pkg/domain/infra/abi/play.go:293]: please use proper error checking with `errors.Is()` and not string comparison.

Comment (flouthoc) [reply pkg/domain/infra/abi/play.go]: done.

**Comment by rhatdan** [comment]
LGTM

**Comment by flouthoc** [comment]
@Luap99 Yes we can do get first and create new volume later, I'll open a follow-up.

**Comment by Luap99** [comment]
No that is not what I am saying. I just want that the newly added if else logic is inverted. This way you can drop the else because you would return in the if block.

**Pull request merged**